x/service/types/message_add_service.go

```diff
@@ -0,0 +1,36 @@
+package types
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/pokt-network/poktroll/x/shared/types"
+)
+
+var _ sdk.Msg = (*MsgAddService)(nil)
+
+func NewMsgAddService(address, serviceId, serviceName string) *MsgAddService {
+	return &MsgAddService{
+		Address: address,
+		Service: types.Service{
+			Id:   serviceId,
+			Name: serviceName,
+		},
+	}
+}
+
+// ValidateBasic performs basic validation of the message and its fields
+func (msg *MsgAddService) ValidateBasic() error {
+	if _, err := sdk.AccAddressFromBech32(msg.Address); err != nil {
+		return ErrServiceInvalidAddress.Wrapf(
```

Comment:
`// TODO_TECHDEBT(#348): ensure consistent error convention is used.`

(below as well)